oak-doc/src/site/markdown/dos_and_donts.md

```diff
@@ -130,3 +130,19 @@ c = d.getParent();                              // preferred way to fetch the pa
 this is that Oak internally uses various classes from the `nio` package that implement 
 `InterruptibleChannel`, which are [asynchronously closed](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/channels/InterruptibleChannel.html) 
 when receiving an `InterruptedException` while blocked on IO. See [OAK-2609](https://issues.apache.org/jira/browse/OAK-2609).  
+
+### Avoid or minimize conflicts
+To reduce the possiblity of having errors like `OakState0001: Unresolved conflicts in ...`:
+
+1. Try to keep the JCR sessions as short as possible. i.e. create the session, make your changes, call session.save() and then session.logout(). If you need to do 
+something additional in the repository, a few lines after (maybe after some processing that could take some time), create the session again and repeat.
+
+2. Try to use session.refresh(true) before saving, if you think that some significant time can pass between the login() and the save() call.
```

Comment:
i would recommend tocall session.refresh(false) before making changes to avoid conflicts. if that is not feasible, session.refresh(true) can be called before the save call.